tools/serverpod_cli/lib/src/analyzer/dart/endpoint_analyzers/endpoint_class_analyzer.dart

```diff
@@ -75,4 +90,22 @@ abstract class EndpointClassAnalyzer {
 
     return fileDirPathParts;
   }
+
+  /// Checks for duplicate class names in a list of [EndpointDefinition] objects.
+  /// Returns the first duplicate class name found, or `null` if no duplicates are found.
+  static String? _checkForDuplicateClassNames(
+    List<EndpointDefinition> endpointDefs,
+  ) {
+    Set<String> classNames = {};
+
+    for (var element in endpointDefs) {
+      // Assuming 'className' is a property of elements in endpointDefs
+      if (classNames.contains(element.className)) {
+        return element.className;
+      }
+      classNames.add(element.className);
+    }
+
+    return null;
+  }
```

Comment:
chore: Change so that it only checks if the current class has a duplicate in some list.

This will always fail for all classes, even if the class analyzed is unique in its name. The `validate` method should report these issues individually for each class analyzed.